rethinkdb_exporter.go

```diff
@@ -351,7 +355,22 @@ func main() {
 		log.Fatal("need parameter addr with len > 0 to connect to RethinkDB cluster")
 	}
 
-	exporter := NewRethinkDBExporter(*addr, *auth, *user, *pass, *clusterName, *namespace)
+	var tlsConfig *tls.Config
+	if *tlsEnable {
+		var err error
+		tlsConfig, err = prepareTLSConfig(*caFile, *certFile, *keyFile)
+		if err != nil {
+			log.Fatalf("failed to prepare tls config: %v", err)
+		}
+	}
+
+	rconn, err := connectRethinkdb(*addr, *auth, *user, *pass, tlsConfig)
```

Comment:
This is problematic as this will try to connect to Rethink at startup (as opposed to scrape time as right now) so if your DB is down when the exporter starts the exporter will fail to come up.

Reply:
fixed with lazy initialization